### PR TITLE
Support user defined delimiter when parsing list

### PR DESCRIPTION
**Pull request by rafiw**
Currently user can only parse list with spaces
E.g app -o 1 2 3
now use can define his own one char delimiter e.g comma

### CODE REVIEW

**Comment by henryiii** [comment]
Please add a few tests, including at least the following cases:

```
-v 1 2
-v 1,2
-v 1,2,
-v 1, 2
-v "1,2"
-v "1, 2 "
```

I believe space will always be a separator, since this is what gets passed to CLI11 from argc, argv.

**Comment by henryiii** [comment]
I can fix the style pretty trivially if needed. Is there a version with defaults to add too?

File: tests/AppTest.cpp [hunk 1]
```diff
@@ -339,7 +339,7 @@ TEST_F(TApp, doubleVectorFunction) {
     });
     args = {"--val", "5", "--val", "6", "--val", "7"};
     run();
-    EXPECT_EQ(res.size(), 3);
+    EXPECT_EQ(res.size(), (size_t)3);
```
Comment (harrywong) [inline tests/AppTest.cpp]: I think `EXPECT_EQ(res.size(), 3u);` is better here :)

Comment (henryiii) [reply tests/AppTest.cpp]: That might be a good idea generally, but for now, this is consistent.

Comment (henryiii) [reply tests/AppTest.cpp]: Okay, this can be changed now. I've converted to using `#u` everywhere in testing. Thanks!

**Comment by henryiii** [comment]
This should be added in the 

```cpp
Option *add_option(std::string option_name,
                       std::vector<T> &variable, ///< The variable vector to set
                       std::string description,
                       bool defaulted)
```
 one too, right below, and matching tests copied and pasted in.

**Comment by rafiw** [comment]
i will update next week

**Comment by rafiw** [comment]
@henryiii there are some style issues not related to my changes, besides that i think it's ok

**Comment by henryiii** [comment]
Okay, I'll rerun the formatter soon.

**Comment by henryiii** [comment]
Looks like the 5.0.1 version of clang-format is a bit finicky too. I'll play with cleanup tomorrow, and will probably add a couple of clang-format-off directives (or switch to a docker clang-format on Travis and use unibeatify's version).

**Comment by henryiii** [comment]
By the way, I believe the API will change a bit with #240: Instead of an extra argument in add_option, you would do it like this now:

```cpp
add_option("--thing", vector)->delimiter(',');
```

As a bonus, the delimiter is now supported by validators and more.

**Comment by rafiw** [comment]
Great work

**Pull request merged**